src/main/java/org/minions/devfund/lourdes/battleship/Ship.java

```diff
@@ -0,0 +1,149 @@
+package org.minions.devfund.lourdes.battleship;
+
+public abstract class Ship {
+    private int bowRow;
+    private int bowColumn;
+    protected int length;
+    private boolean horizontal;
+    protected boolean[] hit;
+
+    abstract String getShipType();
+
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, Ocean ocean) {
+        if (row < 0 || row > ocean.getShipArray().length || column < 0 || column > ocean.getShipArray().length) {
+            return false;
+        }
+        return (verifyShipSpaceFree(row, column, horizontal, ocean, length) && verifyShipBorderFree(row, column, horizontal, ocean));
+    }
+
+    private boolean verifyShipBorderFree(int row, int column, boolean horizontal, Ocean ocean) {
+        boolean topRow = true;
+        boolean bottomRow = true;
+        boolean rightRow = true;
+        boolean leftRow = true;
+        int sizeBorder = horizontal? length + 2 : row + length;
+        if (row > 0 && row < ocean.getShipArray().length && column > 0 && column < ocean.getShipArray().length) {
+            if (horizontal) {
+                topRow = verifyShipSpaceFree(row - 1, column - 1, horizontal, ocean, sizeBorder);
+                bottomRow = verifyShipSpaceFree(row + 1, column - 1, horizontal, ocean, sizeBorder);
+                rightRow = !ocean.isOccupied(row, length + 2);
+                leftRow = !ocean.isOccupied(row, column - 1);
+            } else {
+                topRow = !ocean.isOccupied(row - 1, column);
+                bottomRow = !ocean.isOccupied(row + length, column);
+                rightRow = verifyShipSpaceFree(row - 1, column - 1, horizontal, ocean, sizeBorder);
+                leftRow = verifyShipSpaceFree(row - 1, column + 1, horizontal, ocean, sizeBorder);
+            }
+
+        }
+
+        return topRow && bottomRow && rightRow && leftRow;
+    }
+
+    private boolean verifyShipSpaceFree(int row, int column, boolean horizontal, Ocean ocean, int sizeShip) {
+        try {
+            for (int i = 0; i < sizeShip; i++) {
+                if (horizontal && ocean.isOccupied(row, column + i)) {
+                    return false;
+
+                } else if ((ocean.isOccupied(row + i, column))) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This branch's code block is the same as the block for the branch on line 46. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1871)